processing/download_hydx.py

```diff
@@ -0,0 +1,75 @@
+from pathlib import Path
+import requests
+import time
+from typing import List
+import zipfile
+import io
+
+
+class MockQgsProcessingFeedback:
+    def pushInfo(self, info: str):
+        pass
+
+    def isCanceled(self) -> bool:
+        return False
+
+
+def download_hydx(dataset_name: str, target_directory: Path, wait_times: List = None, feedback=None):
+    """
+    Download GWSW HydX dataset of given `dataset_name` to given `target_directory`. A request to prepare the dataset for
+    download will be send to the GWSW server. Subsequently, status requests will be sent until the dataset is ready for
+    download. The wait time between these requests is determined by `wait_times`, a list of time durations (in seconds).
+    If all `wait_times` have been used up, the last item in the list will be reused until the process has been completed
+    or until the user cancels it (i.e. when feedback.isCanceled() returns True).
+
+    `feedback` should have the methods .pushInfo() and .isCanceled(), e.g. a QgsProcessingFeedback
+
+    :param dataset_name:
+    :param target_directory:
+    :param wait_times:
+    :param feedback:
+    :return:
+    """
+    wait_times = wait_times or [1]
+    feedback = feedback or MockQgsProcessingFeedback()
+    request_hydx_url = f"https://apps.gwsw.nl/hyd/makehydx?filter={dataset_name}&delim=pk"
+    hydx_headers = {"Authorization": "Bearer requestN&S"}
```

Comment:
What is this? "requestN&S" is the authorization header? (is this a security problem?)

For sure constants like these should be defined at the top of the file, you might want to configure that.

Reply:
This is the login key. A security issue I can see here is that other people could access the GWSW API using this key, without our consent. I asked Rioned if they have any other way to organize this, but this is their only solution at the moment. Do you think this security risk is acceptable?

I have now defined this as a global variable (KEY) at the top of the file.

Reply:
So we let any user of ThreeDiToolbox authenticate as N&S on app.gwsw.nl ?  Also, this is public, so anyone can now already authenticate as N&S.

Apparently Rioned expects users to authenticate. Maybe there are limits or they just want to keep statistics. The logical solution is that if a user wants data from app.gwsw.nl, he/she needs to get its own authorization key. Is that something that Rioned could support? I expect this to be an input field in the download tool.

Reply:
I think the security risk is for Rioned, not for us. I asked them about this and they suggested this approach. They don't have any infrastructure for users to create there own key atm.